logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterClientUseCase.kt

```diff
@@ -2,25 +2,26 @@ package com.wire.kalium.logic.feature.client
 
 import com.wire.kalium.logic.data.client.ClientRepository
 import com.wire.kalium.logic.data.client.RegisterClientParam
-import com.wire.kalium.logic.failure.AuthenticationFailure
-import com.wire.kalium.logic.functional.Either
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flow
+import com.wire.kalium.logic.failure.WrongPassword
+import com.wire.kalium.logic.functional.suspending
 
 
-class RegisterClientUseCase(
-    private val clientRepository: ClientRepository
-) {
-    suspend operator fun invoke(param: RegisterClientParam): Flow<RegisterClientResult> = flow {
-        when (val result = clientRepository.registerClient(param)) {
-            is Either.Right -> emit(RegisterClientResult.Success(result.value))
+class RegisterClientUseCase(private val clientRepository: ClientRepository) {
 
-            is Either.Left -> {
-                if (result.value is AuthenticationFailure)
-                    emit(RegisterClientResult.Failure.InvalidCredentials)
-                else
-                    emit(RegisterClientResult.Failure.Generic(result.value))
+    suspend operator fun invoke(param: RegisterClientParam): RegisterClientResult = suspending {
+        //TODO Should we fail here if the client is already registered?
```

Comment:
For sure it doesn't make sense to register the client again but then again wouldn't the BE respond with an error if you attempt register an already registered client?

Reply:
The backend allows you to use the same authentication token to register multiple clients.

We just send the information needed to create/register a new client. And it either registers it - responding with a new ClientId and so on - or fails is some data is not as it expects.

It's another philosophical question. Should we allow multiple clients for the same user session in this SDK?
Is that a valid use case? I kinda think no.

So if the session has an ClientId, we could just fail the registration and return `wtf dude?`.

Reply:
should the check if an account already exist in the app be done here, after login (since login without client registration is possible) or why not both ?

 I will assume that BE returns a new `ClientId` after client register, so is it possible to compare clientIds ?

Reply:
@MohamadJaara Yes, it would be a new `ClientId`. But comparing after registering is too late, either the user created an unnecessary client, which is annoying for cryptographic reasons and might reach the max clients limit, or this could fail unnecessarily.

Considering that the user needs a password in order to register a client, the app asking again for registration is a bit weird and probably a bug. So we checking this in Kalium may prevent a buggy app from re-registering without necessity.

I feel like this check doesn't need to be added right now, but it's a good reminder we can have a say and be opinionated on how the projects relying on this SDK should behave.